### PR TITLE
feat(svg-img-alt): rule for when svg needs a title

### PR DESCRIPTION
**Pull request by WilcoFiers**
This new svg-img-alt rule checks that the `svg` elements with `role=graphics-document`, and any element in the SVG namespace that has `role=img` or `role=graphics-symbol` has an accessible name, either through ARIA label, the `title` attribute or by having a non-empty child `title` element.

This rule limits the role-img-alt rule to elements in the HTML namespace to avoid overlap with the svg-img-alt rule.

Closes issue: #20

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**

- [x] Follows the commit message policy, appropriate for next version
- [x] Code is reviewed for security

### CODE REVIEW

File: lib/rules/role-img-alt.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
 	"id": "role-img-alt",
-	"selector": "[role='img']:not(svg):not(img):not(area):not(input):not(object)",
+	"selector": "[role='img']:not(img):not(area):not(input):not(object)",
```
Comment (straker) [inline lib/rules/role-img-alt.json]: Should add a test to this rules integration test to ensure svg elements aren't captured by this rule now.

Comment (WilcoFiers) [reply lib/rules/role-img-alt.json]: gd catch.

File: lib/checks/shared/svg-non-empty-title.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+const titleNode = virtualNode.children.find(({ props }) => {
+	return props.nodeName === 'title';
```
Comment (jkodu) [inline lib/checks/shared/svg-non-empty-title.js]: I know this is a nitpick, but don't we always go `nodeName.toUpperCase`?

Comment (straker) [reply lib/checks/shared/svg-non-empty-title.js]: The `VirtualNode` interface does that for us now https://github.com/dequelabs/axe-core/blob/develop/lib/core/base/virtual-node/virtual-node.js#L27-L36

**Pull request merged**